### ISSUE TITLE
Last Value in Stream, waiting to be pushed

### ISSUE THREAD

**Issue by pawan4work** [influxdata/kapacitor#1937]
@nathanielc had explained Kapacitor stream behavior as -

**That last value is not skipped its just still sitting in the window waiting to be pushed out by then next point behind it.**    from https://github.com/influxdata/kapacitor/issues/221

you have mentioned, implementing flush is feasible for valid use case.

1/  **logparser**  :  using logparser to push only ERROR messages in Influx, tick job expected to raise an alert on each measurement on STREAM as "logparser" inserting only ERROR lines.

since, each line in log file is not an ERROR message i.e. data is not being inserted every seconds or 10S window,
TICK script is raising alert on previous LAST Message only when next point arrives. This is delaying  alerting on LAST message sitting on Window.

Please suggest how can we achieve this  ?  Is there any other way alerts can be trigger

**Comment by voroninman** [comment]
Is there a way to create a default stream (don't confuse with default field) with an "every" property so the problem in the issue could be addressed by combining two: an actual windowed stream and a default (tick?) stream?

**Comment by voroninman** [comment]
@pawan4work FYI, I ended up with applying WhereNode after the WindowNode so it always has data points to check the time from. Hope that helps.

**Comment by pawan4work** [comment]
@voroninman yeah, this work around am using where ever possible. But case where you are relying on LAST value in stream it won't work. Either some force flush kind of mechanism will help to handle such cases.

e.g.
1. InfluxOut writing data in some measurement and alerts on other script based on that last entered record in measurement 

2. 'exec' plug-in insert some last failed results, it won't be get captured in TICK script until plug-in insert additional two or more failures status.

**Comment by sergolius** [comment]
@pawan4work can [BarrierNode](https://docs.influxdata.com/kapacitor/v1.5/nodes/barrier_node/) help you in this case? `idle` method "pushes" pipeline `on idle time since the last received message`.
